runners/spark/src/main/java/org/apache/beam/runners/spark/translation/streaming/SparkRunnerStreamingContextFactory.java

```diff
@@ -95,6 +96,25 @@ public JavaStreamingContext create() {
     return jssc;
   }
 
+  private void checkpoint(JavaStreamingContext jssc) {
+    Path rootCheckpointPath = checkpointDir.getRootCheckpointDir();
+    Path sparkCheckpointPath = checkpointDir.getSparkCheckpointDir();
+
+    try {
+      FileSystem fileSystem = rootCheckpointPath.getFileSystem(jssc.sc().hadoopConfiguration());
```

Comment:
Comment: `// we separate Spark's checkpointing from the runner's checkpoint into different directories`.